src/tests/test_circumcision.py

```diff
@@ -0,0 +1,435 @@
+                               & (pop.data[col.AGE] < 50)]
+    prob_circ = (pop.date.year - pop.circumcision.mc_int) * pop.circumcision.circ_inc_rate
+    mean = len(male_population) * prob_circ
+    stdev = sqrt(mean * (1 - prob_circ))
+    # basic checks
+    general_circumcision_checks(mean, stdev, no_vmmc, pop.data)
+    # nobody over 50 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] >= 50) & (pop.data[col.VMMC])]) == 0
+
+    # evolve population
+    while pop.date <= stop_date:
+        circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male) & pop.data[col.CIRCUMCISED]]
+        # advance ages and vmmc
+        pop.data.age += time_step.days / 365
+        pop.circumcision.update_vmmc(pop)
+        pop.date += time_step
+        # check circumcisied people remain circumcised each step
+        new_circ_males = pop.data.index[(pop.data[col.SEX] == SexType.Male)
+                                        & pop.data[col.CIRCUMCISED]]
+        assert circ_males.isin(new_circ_males).all()
+
+    # nobody under 10 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] < 10) & (pop.data[col.VMMC])]) == 0
+
+
+def test_vmmc_case_i1():
+
+    N = 100000
+    start_date = date(2022, 1, 1)
+    time_step = timedelta(days=90)
+
+    # build population
+    pop = Population(size=N, start_date=start_date)
+    pop.data[col.SEX] = SexType.Male
+    pop.data[col.CIRCUMCISED] = False
+    pop.data[col.CIRCUMCISION_DATE] = None
+    pop.data[col.VMMC] = False
+
+    # case 1
+    # circumcision stops in 10-14 year olds and
+    # increases in 15-19 year olds after year_interv
+    pop.circumcision.circ_inc_rate_year_i = 1
+    pop.circumcision.covid_disrup_affected = 0
+    pop.circumcision.vmmc_disrup_covid = 0
+    pop.circumcision.mc_int = 2008
+    pop.circumcision.year_interv = 2022
+
+    # evolve population
+    pop.data.age += time_step.days / 365
+    pop.circumcision.update_vmmc(pop)
+    # nobody under 15 has been circumcised
+    assert len(pop.data[(pop.data[col.AGE] < 15) & (pop.data[col.VMMC])]) == 0
+
+    # get stats
+    no_vmmc = len(pop.data[pop.data[col.VMMC]
+                           & (pop.data[col.AGE] >= 15)
+                           & (pop.data[col.AGE] < 20)])
+    male_population = pop.data[(pop.data[col.SEX] == SexType.Male)
+                               & (pop.data[col.AGE] >= 15)
+                               & (pop.data[col.AGE] < 20)]
+    prob_circ = ((2013 - pop.circumcision.mc_int) + (2019 - 2013)
+                 * pop.circumcision.rel_incr_circ_post_2013
+                 * pop.circumcision.circ_inc_15_19) * pop.circumcision.circ_inc_rate
+    # cap probability at 1
+    if (prob_circ > 1):
+        prob_circ = 1
+    mean = len(male_population) * prob_circ
+    stdev = sqrt(mean * (1 - prob_circ))
+    # check circumcised value is within 3 standard deviations
+    assert mean - 3 * stdev <= no_vmmc <= mean + 3 * stdev
```

Comment:
Is there a reason not to use `general_circumcision_checks` here and for other vmmc tests?

Reply:
The only reason I didn't was because I expect `test_vmmc_basic` (acting as my `test_vmmc_case_i0`) to catch if anything untoward happens in the general checks. The population is entirely male so there's no reason to check if any females have been marked as circumcised, and I expect that if not all circumcised people have circumcision dates (or if some uncircumcised people have circumcision dates) this issue will have been caught earlier. I could use `general_circumcision_check`, I just think most of the checks are redundant.